store/sqlstore/post_store.go

```diff
@@ -1580,26 +1585,34 @@ func (s *SqlPostStore) GetPostAfterTime(channelId string, time int64, collapsedT
 	return &post, nil
 }
 
-func (s *SqlPostStore) getRootPosts(channelId string, offset int, limit int, skipFetchThreads bool) ([]*model.Post, error) {
+func (s *SqlPostStore) getRootPosts(channelId string, offset int, limit int, skipFetchThreads bool, includeDeleted bool) ([]*model.Post, error) {
 	posts := []*model.Post{}
 	var fetchQuery string
 	if skipFetchThreads {
 		fetchQuery = "SELECT p.*, (SELECT COUNT(*) FROM Posts WHERE Posts.RootId = (CASE WHEN p.RootId = '' THEN p.Id ELSE p.RootId END) AND Posts.DeleteAt = 0) as ReplyCount FROM Posts p WHERE p.ChannelId = ? AND p.DeleteAt = 0 ORDER BY p.CreateAt DESC LIMIT ? OFFSET ?"
 	} else {
 		fetchQuery = "SELECT * FROM Posts WHERE Posts.ChannelId = ? AND Posts.DeleteAt = 0 ORDER BY Posts.CreateAt DESC LIMIT ? OFFSET ?"
 	}
+	if includeDeleted {
+		fetchQuery = "SELECT * FROM Posts WHERE Posts.ChannelId = ? ORDER BY Posts.CreateAt DESC LIMIT ? OFFSET ?"
+	}
```

Comment:
I'm not sure this is right -  here `includeDeleted` is overwriting the `skipFetchThreads` conditions.
If the caller passes in both `includeDeleted` and `skipFetchThreads` as true, then we won't get the expected "ReplyCount" column.

I think the `includeDeleted` needs to apply to the `skipFetchThreads` as well.

Reply:
So something like

```Go
var fetchQuery string
if skipFetchThreads {
    fetchQuery = "....." // query for skipFetchThreads
    if !includeDeleted {
        fetchQuery += "WHERE DeleteAt = 0" // update conditions to not return deleted posts
    }
} else {
    fetchQuery = "....." // query if not skipFetchThreads
    if !includeDeleted {
        fetchQuery += "WHERE DeleteAt = 0" // update conditions to not return deleted posts
    }
}

Reply:
For some reason, I understood that the `include_deleted` was an independent parameter but you are correct it needs to apply to `skipFetchThreads` as well.

Reply:
Done